index.js

```diff
@@ -89,22 +119,24 @@ function storeOptionValue(options, longOption, value, result) {
     // result.values[longOption] starts out not present,
     // first value is added as new array [newValue],
     // subsequent values are pushed to existing array.
-    const usedAsFlag = value === undefined;
-    const newValue = usedAsFlag ? true : value;
+    const usedAsFlag = optionValue === undefined;
+    const newValue = usedAsFlag ? true : optionValue;
     if (result.values[longOption] !== undefined)
       ArrayPrototypePush(result.values[longOption], newValue);
     else
       result.values[longOption] = [newValue];
   } else {
-    result.values[longOption] = value;
+    result.values[longOption] = optionValue;
   }
 }
 
 const parseArgs = ({
   args = getMainArgs(),
+  strict = false,
```

Comment:
Let's default this to `true` and update tests accordingly.

Reply:
Started with a [naive approach](https://github.com/pkgjs/parseargs/pull/74/commits/e66a8b35714b693a200fe5dc2b3fde9541c3ae42) to get an understanding of how this change affects our existing tests:
- Defaulted `parseArgs` to `strict:true`
- Updated only failing tests to `strict:false`

I'm committing early for visibility but also I will likely need assistance updating/validating these tests. At first glance of the diff I'm wondering if we can clear things up by organizing these tests in a `strict-false.js` file or grouping `strict:false` tests in each file. I'm also fine with scoping that out to another PR.

cc @shadowspawn